test/unit_overapproximate.jl

```diff
@@ -165,6 +165,24 @@ for N in [Float64, Float32]
     Y_zonotope = overapproximate(Y, Zonotope) # overapproximate with a zonotope
     @test Y_polygon ⊆ Y_zonotope
     @test !(Y_zonotope ⊆ Y_polygon)
+
+    #decomposed intersection between cartesian product array and abstract polyhedron
+    i1 = Interval(N[0, 1])
+    i2 = Interval(N[2, 3])
+    h1 = Hyperrectangle(low=N[3, 4], high=N[5, 7])
+    h2 = Hyperrectangle(low=N[5, 5], high=N[6, 8])
+    cpa1 = CartesianProductArray([i1, i2, h1])
+    cpa2 = CartesianProductArray([i1, i2, h2])
+    G = HPolyhedron([HalfSpace(N[1, 0, 0, 0], N(1))])
+    G_empty = HPolyhedron([HalfSpace(N[1, 0, 0, 0], N(-1))])
+
+    d_int_g = Intersection(cpa1, G)
+    l_int_g = Intersection(cpa1, G)
+    d_int_cpa = intersection(cpa1, cpa2)
+    l_int_cpa = Intersection(cpa1, cpa2)
```

Comment:
`d_int_g == l_int_g` and `d_int_cpa == l_int_cpa`!

Reply:
Do you mean that they are equal and I can just remove duplication?

Reply:
I do not know what you intended to test, so this is just the observation that they are equal and that I have the feeling that you may not have wanted this.

Reply:
I removed duplication. The idea is that I am calling different `overapproximate`. Old one and new decomposed version. Then I am checking whether results are the same.